phoenix-core/src/main/java/org/apache/phoenix/coprocessor/metrics/MetricsPhoenixCoprocessorSourceFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.apache.phoenix.coprocessor.metrics;
+
+public class MetricsPhoenixCoprocessorSourceFactory {
+
+    private static final MetricsPhoenixCoprocessorSourceFactory
+            INSTANCE = new MetricsPhoenixCoprocessorSourceFactory();
+    private MetricsPhoenixTTLSource phoenixTTLSource;
+
+    public static MetricsPhoenixCoprocessorSourceFactory getInstance() {
```

Comment:
When you change to lazily initializing instead, we would need to make this thread-safe.